### PR TITLE
@l2succes => Canvas-based Thumbnail Image

### PR DESCRIPTION
**Pull request by kanaabe**
This uses the Canvas API to "draw" some custom text to a canvas, generate an image out of it, converts it to a Blob type, and uploads it to S3 using `gemup`.

The screenshots below show a mismatch in cropping, but that's actually intentional. This format is optimized for social which is seen in the social tab as I scroll down. 

Generates Image:
canvas-image-generator

Can use scheduled publish date:
canvas-image-generator-2

### CODE REVIEW

**Comment by orta** [comment]
Cool 👍

**Comment by l2succes** [comment]
Nice! Is the plan to use the article image for the background or it's meant to remain a black?

File: src/client/components/image_generator/index.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+interface Props {
+  article: any
+  onChangeArticleAction: any
+}
+
+export class ImageGenerator extends Component<Props, State> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
```
Comment (l2succes) [inline src/client/components/image_generator/index.tsx]: you can also do 

```typescript
export class ImageGenerator extends Component<Props, State> {
  state = {
      text: this.props.article.thumbnail_title || ""
  }
...

**Comment by kanaabe** [comment]
@l2succes We got new designs and they're all black background!

File: src/client/components/image_generator/index.tsx (hunk 1 above)
Comment (kanaabe) [reply src/client/components/image_generator/index.tsx]: 👍 Good catch

**Comment by kanaabe** [comment]
@l2succes any more comments on this one? They wanted to start QA'ing by EOD.

**Comment by l2succes** [comment]
@kanaabe Nope! Looks good!

**Pull request merged**